code/modules/power/singularity/emitter.dm

```diff
@@ -175,81 +198,94 @@
-		switch(state)
-			if(EMITTER_LOOSE)
-				to_chat(user, SPAN_WARNING("\The [src] needs to be wrenched to the floor."))
-			if(EMITTER_BOLTED)
-				if(WT.remove_fuel(0, user))
-					playsound(get_turf(src), 'sound/items/Welder2.ogg', 50, TRUE)
-					user.visible_message(SPAN_NOTICE("\The [user] starts to weld \the [src] to the floor."), \
-						SPAN_NOTICE("You start to weld \the [src] to the floor."), \
-						SPAN_WARNING("You hear the sound of metal being welded."))
-					if(do_after(user, 20 / W.toolspeed, act_target = src))
-						if(!src || !WT.isOn())
-							return
-						state = EMITTER_WELDED
-						to_chat(user, SPAN_NOTICE("You weld \the [src] to the floor."))
-						connect_to_network()
-				else
-					to_chat(user, SPAN_WARNING("You need more welding fuel to complete this task."))
-			if(EMITTER_WELDED)
-				if(WT.remove_fuel(0, user))
-					playsound(get_turf(src), 'sound/items/Welder2.ogg', 50, TRUE)
-					user.visible_message(SPAN_NOTICE("\The [user] starts to cut \the [src] free from the floor."), \
-						SPAN_NOTICE("You start to cut \the [src] free from the floor."), \
-						SPAN_WARNING("You hear the sound of metal being welded."))
-					if(do_after(user, 20 / W.toolspeed, act_target = src))
-						if(!src || !WT.isOn())
-							return
-						state = EMITTER_BOLTED
-						to_chat(user, SPAN_NOTICE("You cut \the [src] free from the floor."))
-						disconnect_from_network()
-				else
-					to_chat(user, SPAN_WARNING("You need more welding fuel to complete this task."))
+		var/obj/item/emitter_barrel/EB = W
+		if(EB.build_state != BARREL_UNMODIFIED)
+			to_chat(user, SPAN_WARNING("\The [EB] has auxiliary attachments and can't fit into the [src]!"))
+			return
+		user.visible_message(SPAN_NOTICE("\The [user] installs \the [EB] into \the [src]."), SPAN_NOTICE("You install \the [EB] into \the [src]."), range = 3)
+		user.drop_from_inventory(EB, src)
+		barrel = EB
 		return
 
 	if(istype(W, /obj/item/card/id) || istype(W, /obj/item/device/pda))
 		if(emagged)
 			to_chat(user, SPAN_WARNING("The lock seems to be broken."))
 			return
 		if(allowed(user))
-			if(active)
-				locked = !locked
-				to_chat(user, SPAN_NOTICE("The controls are now [locked ? "locked." : "unlocked."]"))
-			else
-				locked = FALSE //just in case it somehow gets locked
-				to_chat(user, SPAN_WARNING("The controls can only be locked when \the [src] is online."))
+			locked = !locked
+			to_chat(user, SPAN_NOTICE("The controls are now [locked ? "locked." : "unlocked."]"))
 		else
 			to_chat(user, SPAN_WARNING("Access denied."))
 		return
+
+	if(W.isscrewdriver())
+		if(!barrel)
+			to_chat(user, SPAN_WARNING("\The [src] doesn't have a barrel!"))
+			return
+		barrel_secured = !barrel_secured
+		var/others_msg = "\The [user] [barrel_secured ? "" : "un"]secures \the [src]'s barrel."
+		var/self_msg = "You [barrel_secured ? "" : "un"]secure \the [src]'s barrel."
+		user.visible_message(SPAN_WARNING(others_msg), SPAN_WARNING(self_msg), range = 3)
```

Comment:
would appreciate it if this followed this convention: bold the name with no spans for others, leave as is for self

Reply:
this applies to like every visible message here